### PR TITLE
CLI flag for disabling interrupts

### PR DESCRIPTION
**Pull request by parsonsmatt**
This PR adds a cli flag that disables the interrupt behavior for reloads. This is my first contribution to a Rust project so I'm sure it'll take some polishing :)

### CODE REVIEW

File: src/cli.rs [hunk 1]
```diff
@@ -49,6 +49,11 @@ pub struct Opts {
     /// Options to modify logging and error-handling behavior.
     #[command(flatten)]
     pub logging: LoggingOpts,
+
+    /// By default, ghciwatch will interrupt reloads if a file changes. If you want ghciwatch to
+    /// avoid interrupting reloads, set this flag.
+    #[arg(long = "no-interrupt-reloads")]
+    pub no_interrupt_reloads: bool,
```
Comment (parsonsmatt) [inline src/cli.rs:56]: I'm not super happy with a `--no-interrupt-reloads` flag - feels kind of awkward. But `--interrupt-reloads` is the default behavior.

We could switch the default back to `no-interrupt`, and then this flag can be `--interrupt-reloads`.

Made a slack convo to ask about this: 
https://mercurytechnologies.slack.com/archives/CUJ89R895/p1702483630925389

File: src/ghci/manager.rs [hunk 2]
```diff
@@ -56,6 +56,7 @@ pub async fn run_ghci(
     // This function is pretty tricky! We need to handle shutdowns at each stage, and the process
     // is a little different each time, so the `select!`s can't be consolidated.
 
+    let no_interrupt_reloads = opts.no_interrupt_reloads;
```
Comment (parsonsmatt) [inline src/ghci/manager.rs:59]: This definitely feels awkward. I tried passing the `opts` directly in to the relevant function, but the `opts` are moved into `Ghci::new` call just below. I can get the `bool` out here no problem, though, so I do that instead.

File: src/ghci/manager.rs [hunk 3]
```diff
@@ -113,19 +114,21 @@ pub async fn run_ghci(
...
-                    event.merge(new_event);
-                    maybe_event = Some(event);
-
-                    // Cancel the in-progress reload. This releases the `ghci` lock to prevent a deadlock.
-                    task.abort();
-
-                    // Send a SIGINT to interrupt the reload.
-                    // NB: This may take a couple seconds to register.
-                    ghci.lock().await.send_sigint().await?;
+                if !no_interrupt_reloads {
```
Comment (parsonsmatt) [inline src/ghci/manager.rs:117]: here's our behavior change: if we set this flag, then we do not go into that block

File: src/cli.rs (hunk 1 above)
Comment (9999years) [reply src/cli.rs:56]: There's an open issue for automatic negation flags (i.e. adding `--interrupt-reloads` and `--no-interrupt-reloads` at once) here:

https://github.com/clap-rs/clap/issues/815

File: src/ghci/manager.rs (hunk 2 above)
Comment (9999years) [reply src/ghci/manager.rs:59]: Yeah I think this is fine.

**Review by 9999years — APPROVED** [review]

**Pull request merged**